public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,231 @@
+  let filterValid = filterInfo.getInstance({
+    filterExpression: config.filterExpression,
+    fieldType: fieldType,
+  }).isValid;
+
+  let fieldNameInvalid =
+    config.fieldName === null ||
+    fieldNameOptions.filter((item: Record<string, any>) => item.value === config.fieldName).length === 0;
+  let filterTypeInvalid =
+    !fieldNameInvalid && filterInfo.supportedFieldTypes && !filterInfo.supportedFieldTypes.includes(fieldType);
+  let filterExpressionInvalid = !fieldNameInvalid && !filterTypeInvalid && !filterValid;
+
+  let filterOptionsInput = null;
+  if (filterInfo.placeholder) {
+    filterOptionsInput = (
+      <>
+        <Input
+          className="flex-grow-1"
+          invalid={filterExpressionInvalid}
+          defaultValue={config.filterExpression}
+          placeholder={filterInfo.placeholder}
+          onBlur={event => {
+            onConfigChange({ ...config, filterExpression: event.currentTarget.value });
+          }}
+        />
+      </>
+    );
+  }
+
+  return (
+    <div className="gf-form-inline">
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Filter Type</div>
+        <Select
+          className="width-8"
+          options={[
+            { label: 'Include', value: 'include' },
+            { label: 'Exclude', value: 'exclude' },
+          ]}
+          value={config.type}
+          onChange={option => {
+            // console.log('onChange filterType', option.value);
+            onConfigChange({ ...config, type: option.value });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Filter on Field</div>
+        <Select
+          className="width-16"
+          placeholder="Field Name"
+          options={fieldNameOptions}
+          value={config.fieldName}
+          invalid={fieldNameInvalid}
+          onChange={value => {
+            // console.log('onChange fieldName', value);
+            if (value === null) {
+              onConfigChange({ ...config, fieldName: null });
+            } else {
+              onConfigChange({ ...config, fieldName: value.value });
+            }
+          }}
+          isClearable
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-4">Test</div>
+        <Select
+          invalid={filterTypeInvalid}
+          className="width-8"
+          placeholder="Select test"
+          options={filterTypeOptions}
+          value={config.filterType}
+          onChange={value => {
+            // console.log('onChange test', value);
+            onConfigChange({ ...config, filterType: value.value });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form--grow gf-form-spacing ">{filterOptionsInput}</div>
```

Comment:
```suggestion
      <div className="gf-form gf-form--grow gf-form-spacing ">
        {filterInfo.placeholder && (
          <Input
            className="flex-grow-1"
            invalid={filterExpressionInvalid}
            defaultValue={config.filterExpression}
            placeholder={filterInfo.placeholder}
            onBlur={event => {
              onConfigChange({ ...config, filterExpression: event.currentTarget.value });
            }}
          />
        )}
      </div>
```

I would suggest to do it this way.